### PR TITLE
Validate schema field types based on data store

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/4391

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [x] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: compiler-plugin-test/src/test/resources/project_3/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[package]
+org = "root"
+name = "project_3"
+version = "0.1.0"
+
+[persist]
+datastore = "google-sheets"
```
Comment (daneshk) [inline compiler-plugin-test/src/test/resources/project_3/Ballerina.toml:7]:
```suggestion
datastore = "googlesheets"
```

Comment (daneshk) [reply compiler-plugin-test/src/test/resources/project_3/Ballerina.toml:7]: We decided to remove `-` from the datasource names. please do the change

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/Constants.java [hunk 2]
```diff
@@ -61,4 +63,19 @@ private BallerinaTimeTypes() {
...
     }
 
+    /**
+     * Constants related to Persist datastores.
+     */
+    public static final class Datastores {
+
+        public static final String MYSQL = "mysql";
+        public static final String IN_MEMORY = "in-memory";
+        public static final String GOOGLE_SHEETS = "google-sheets";
```
Comment (daneshk) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/Constants.java:73]:
```suggestion
        public static final String IN_MEMORY = "inmemory";
        public static final String GOOGLE_SHEETS = "googlesheets";
```

Comment (kaneeldias) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/Constants.java]: Fixed in [071d253](https://github.com/ballerina-platform/module-ballerina-persist/pull/196/commits/071d253b42c588d7d79babd70d53b07a2265cb0f)

File: compiler-plugin-test/src/test/resources/project_4/Ballerina.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[package]
+org = "root"
+name = "project_4"
+version = "0.1.0"
+
+[persist]
+datastore = "in-memory"
```
[the remark this answers is not in the document]
Comment (kaneeldias) [reply compiler-plugin-test/src/test/resources/project_4/Ballerina.toml]: Fixed in [071d253](https://github.com/ballerina-platform/module-ballerina-persist/pull/196/commits/071d253b42c588d7d79babd70d53b07a2265cb0f)

File: compiler-plugin-test/src/test/resources/project_3/Ballerina.toml (hunk 1 above)
Comment (kaneeldias) [reply compiler-plugin-test/src/test/resources/project_3/Ballerina.toml]: Fixed in [071d253](https://github.com/ballerina-platform/module-ballerina-persist/pull/196/commits/071d253b42c588d7d79babd70d53b07a2265cb0f)

**Review by daneshk — APPROVED** [review]

**Pull request merged**